04.DOM_manipulation/src/domManipulation.ts

```diff
@@ -0,0 +1,22 @@
+class Dom {
+  remove(element: HTMLElement) {
+    return element.remove();
+  }
+  append(targetElement: HTMLElement, element: HTMLElement) {
+    return targetElement.append(element)
+  }
+  after(targetElement: HTMLElement, element: HTMLElement) {
+    return targetElement.after(element);
+  }
+  prepend(targetElement: HTMLElement, element: HTMLElement) {
+    return targetElement.prepend(element);
+  }
+  before(targetElement: HTMLElement, element: HTMLElement) {
+    return targetElement.before(element);
+  }
+  val(targetElement: HTMLInputElement) {
```

Comment:
what about html elements?

Reply:
Would `HTMLElement & { value: string })` be OK or is there a more elegant solution?

Reply:
is it and or or?

Reply:
I think `and` since it needs to be both an html element and have the `value` property.

Reply:
what would you expect that you get from <div>kreso<div> ?

Reply:
Isn't this `val` method just for the value attribute? Like for `<button value="something">kreso<button>`

Reply:
you're correct - my fault here. sorry :D
but I would like i to behave a bit differently.
value of a input field is the value
value of a HTML element is the text (well also HTML nodes - but let's keep it human readable)   
makes sense? (just so you can play with multiple returns now :D )